sensor/common/clusterentities/store_container_id_test.go

```diff
@@ -0,0 +1,282 @@
+		"Memory disabled with no reset should remember pod1 forever": {
+			numTicksToRemember: 0,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true,
+					},
+				},
+			},
+			operationAfterTick: map[int]operation{}, // do not reset at all
+			containerIDsAfterTick: []map[string]whereThingIsStored{
+				{"pod1": theMap}, // before tick 1: container should be added immediately
+				{"pod1": theMap}, // after tick 1: no reset - should be in the map forever
+				{"pod1": theMap},
+			},
+			publicIPsAtCleanup: []string{},
+		},
+		"Memory disabled with no reset and container overwrite should remember pod1 forever": {
+			numTicksToRemember: 0,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true, // append
+					},
+				},
+				3: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  false, // delete and add
+					},
+				},
+			},
+			operationAfterTick: map[int]operation{}, // do not reset at all
+			containerIDsAfterTick: []map[string]whereThingIsStored{
+				{"pod1": theMap}, // before tick 1: container should be added immediately
+				{"pod1": theMap}, // after tick 1: no reset - should be in the map forever
+				{"pod1": theMap}, // after tick 2
+				// container is overwritten in the map
+				{"pod1": theMap}, // after tick 3: should be still in the map
+				{"pod1": theMap}, // after tick 3: should be still in the map
+			},
+		},
+		"Memory disabled with reset after tick 1 should make pod1 be forgotten before tick 2": {
+			numTicksToRemember: 0,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true,
+					},
+				},
+			},
+			operationAfterTick: map[int]operation{1: mapReset},
+			containerIDsAfterTick: []map[string]whereThingIsStored{
+				{"pod1": theMap}, // before tick 1: container should be added immediately
+				{"pod1": theMap}, // after tick 1: no reset yet, so it should be in the map
+				// reset
+				{"pod1": nowhere}, // after tick 2: should be gone
+			},
+		},
+		"Memory for 2 ticks with reset after tick 1 should make pod1 be forgotten before tick 4": {
+			numTicksToRemember: 2,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true,
+					},
+				},
+			},
+			operationAfterTick: map[int]operation{1: mapReset},
+			containerIDsAfterTick: []map[string]whereThingIsStored{
+				{"pod1": theMap}, // before tick 1: container should be added immediately
+				{"pod1": theMap}, // after tick 1
+				// reset
+				{"pod1": history}, // after tick 2: will remember that for one more tick
+				{"pod1": history}, // after tick 3: will remember that for this last tick
+				{"pod1": nowhere}, // after tick 4: history expired - should be forgotten
+			},
+		},
```

Comment:
The question is about the testcase `"Memory for 2 ticks with reset after tick 1 should make pod1 be forgotten before tick 4"`

#### Answering your comment
That is not so trivial, because there are two cases that we must consider:

1. If the new pod is exactly the same as the old one, then it's data should be either in the current, or in the history but not in both.
2. But imagine that the pod in history was slightly different than the one that is being added again (e.g., different IP, or container ID), then we want to keep that in the history and to some degree have a duplication. If there are lookups for the old value we return from history, however, if the new value is being searched, then we return form the current map.

#### Regarding the context of the comment
You commented on a test case that does not cover re-adds, but there are other test cases that cover exactly what you ask for. Do they make it clearer?